components/App.js

```diff
@@ -42,6 +42,124 @@ export const onclickBindings = {
       menu,
     });
   },
+  'threeD-social-tradeBtn': (e) => {
+    const { menu } = getState();
+    menu.trade.visible = true;
+    menu.trade.toPeer = e.name;
+    menu.trade.fromPeer = loginManager.getAddress();
+    setState({
+      menu,
+    });
+  },
+  'threeD-trade-agreement': (e) => {
+    const { menu } = getState();
+    menu.trade.agreement = !menu.trade.agreement;
+    setState({ menu });
+  },
+  'threeD-trade-header-backBtn': (e) => {
+    const { menu } = getState();
+    menu.trade = {
+      visible: false,
+      toPeer: null,
+      fromPeer: null,
+      selectedItem: null,
+      agreement: false,
+      inventoryPage: 0,
+      peersPage: 0,
+    }
+    setState({
+      menu,
+    });
+  },
+  'threeD-trade-cancel': (e) => {
+    const { menu } = getState();
+    menu.trade = {
+      visible: false,
+      toPeer: null,
+      fromPeer: null,
+      selectedItem: null,
+      agreement: false,
+      inventoryPage: 0,
+      peersPage: 0,
+    }
+    setState({
+      menu,
+    });
+  },
+  'threeD-trade-peers-card': (e) => {
+    console.log('hello', e)
+    const { menu } = getState();
+    menu.trade.toPeer = e.name;
+    setState({ menu });
+  },
+  'threeD-trade-inventory-back': (e) => {
+    const { menu } = getState();
+    menu.trade.inventoryPage > 0 ? --menu.trade.inventoryPage : null;
+    setState({ menu });
+  },
+  'threeD-trade-inventory-forward': (e) => {
+    const { menu } = getState();
+    ++menu.trade.inventoryPage;
+    setState({ menu });
+  },
+  'threeD-trade-peers-back': (e) => {
+    const { menu } = getState();
+    menu.trade.peersPage > 0 ? --menu.trade.peersPage : null;
+    setState({ menu });
+  },
+  'threeD-trade-peers-forward': (e) => {
+    const { menu } = getState();
+    ++menu.trade.peersPage;
+    setState({ menu });
+  },
+  'threeD-trade-accept': async (e) => {
+    const { menu } = getState();
+    if (menu.trade.agreement && menu.trade.toPeer && menu.trade.fromPeer && menu.trade.selectedItem) {
+
+      const trade = {
+        toPeer: menu.trade.toPeer,
+        fromPeer: menu.trade.fromPeer,
+        item: menu.trade.selectedItem
+      };
+
+      const contractSource = await getContractSource('transferNft.cdc');
+      const res = await fetch(`https://accounts.exokit.org/sendTransaction`, {
+        method: 'POST',
+        body: JSON.stringify({
+          address: trade.fromPeer,
+          mnemonic: loginManager.getMnemonic(),
+          limit: 1000,
+          transaction: contractSource
+            .replace(/ARG0/g, trade.item)
+            .replace(/ARG1/g, '0x' + trade.toPeer)
+            .replace(/ARG2/g, 1),
+          wait: true,
+        }),
+      });
+      const response2 = await res.json();
+      console.log(response2)
+
+      menu.trade = {
```

Comment:
This looks like it should be a piecewise assignment, not a replacement.

Otherwise this code and the initialization code could become out of sync.